lib/compat.php

```diff
@@ -74,3 +74,21 @@ function gutenberg_provide_render_callback_with_block_object( $pre_render, $bloc
 	return apply_filters( 'render_block', $block_content, $block );
 }
 add_filter( 'pre_render_block', 'gutenberg_provide_render_callback_with_block_object', 10, 2 );
+
+/**
```

Comment:
Should we have a separate file for template-related PHP logic? This `compat.php` file is at high risk for becoming a dumping ground of various utility functions, so should be limited to functions which can be very closely associated with a specific Trac ticket to incorporate a given change. It doesn't seem like we have such a Trac ticket we could link to with this?

Reply:
Yes, that would be good. Maybe even `template-utils.php`.

Reply:
> Yes, that would be good. Maybe even `template-utils.php`.

Would `template.php` be sufficient to describe the grouping of this related functionality?

Reply:
I think it would be confused with `templates.php`. Maybe a templates directory makes sense at this point? Then, this file could just be called `utils.php`, `templates.php` would become `create-cpt.php` and so on.

Reply:
Oh, I missed that there was a file already. In what way does this function differ that makes it not suitable to be placed in `templates.php` ?

Reply:
`templates.php` just sets up the CPT.

Reply:
This is a utility function for server rendered block implementations that use the post loop.

You could even argue it's not template specific and belongs in `compat.php`.